iceoryx2-ffi/cxx/include/iox2/response_mut.hpp

```diff
@@ -58,37 +62,63 @@ class ResponseMut {
     auto user_header_mut() -> T&;
 
     /// Returns a reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, void>>
     auto payload() const -> const ResponsePayload&;
 
+    /// Returns a reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, void>>
+    auto payload() const -> iox::ImmutableSlice<ValueType>;
+
     /// Returns a mutable reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, void>>
     auto payload_mut() -> ResponsePayload&;
 
-    /// Sends a [`ResponseMut`] to the corresponding
-    /// [`PendingResponse`](crate::pending_response::PendingResponse) of the
-    /// [`Client`](crate::port::client::Client).
-    auto send() -> iox::expected<void, SendError>;
+    /// Returns a mutable reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, void>>
+    auto payload_mut() -> iox::MutableSlice<ValueType>;
 
   private:
     template <ServiceType, typename, typename>
     friend class ResponseMutUninit;
+    template <ServiceType, typename, typename, typename, typename>
+    friend class ActiveRequest;
 
-    explicit ResponseMut();
+    /// Sends a [`ResponseMut`] to the corresponding
+    /// [`PendingResponse`](crate::pending_response::PendingResponse) of the
+    /// [`Client`](crate::port::client::Client).
+    template <ServiceType S, typename ResponsePayloadT, typename ResponseHeaderT>
+    friend auto send(ResponseMut<S, ResponsePayloadT, ResponseHeaderT>&& response) -> iox::expected<void, SendError>;
```

Comment:
Shouldn't this be a public method?

Reply:
See `SampleMut`; we want to move the ownership here. With a public method, this would be more cumbersome.